webserver/webserver/src/main/java/io/helidon/webserver/RequestRouting.java

```diff
@@ -100,13 +110,223 @@ public void route(BareRequest bareRequest, BareResponse bareResponse) {
             RoutedRequest nextRequests = new RoutedRequest(bareRequest, response, webServer, crawler, errorHandlers, span);
             // only register the span context once on the top level request, as others are cloned from it
             nextRequests.context().register(span.context());
-            Contexts.runInContext(nextRequests.context(), (Runnable) nextRequests::next);
+
+            // we must execute system services
+            Contexts.runInContext(nextRequests.context(), () -> {
```

Comment:
If it will be common to have zero system services, perhaps this whole block should be inside a conditional to avoid adding request overhead.

Reply:
There is one line that is always executed `nextRequests.next()`

Reply:
```
List<SystemService> systemServices = webServer.configuration().systemServices();
if (systemServices.isEmpty()) {
    Contexts.runInContext(nextRequests.context(), (Runnable) nextRequests::next);
} else {
    Contexts.runInContext(nextRequests.context(), () -> {
        ServerRequest systemRequest = systemRequest(nextRequests);
        ServerResponse systemResponse = systemResponse(response);
        for (SystemService service : systemServices) {
            service.handle(systemRequest, systemResponse);
        }
        nextRequests.next();
    });
}
```
I understand that the extra complexity is not ideal (and certainly unwarranted if the list will generally be non-empty), but I do worry about unnecessary overhead in request handling slowing us down.

Reply:
ok

Reply:
There is no difference in performance between your proposed solution and my PR. You are doing exactly the same decision, only outside of the `Context.runInContext`